From patchwork Fri Jun 24 12:54:20 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Mel Gorman <mgorman@techsingularity.net>
X-Patchwork-Id: 12894470
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id CDD32C433EF
	for <linux-mm@archiver.kernel.org>; Fri, 24 Jun 2022 12:55:19 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 6CD698E021A; Fri, 24 Jun 2022 08:55:19 -0400 (EDT)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 67DFE8E020E; Fri, 24 Jun 2022 08:55:19 -0400 (EDT)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 56BC18E021A; Fri, 24 Jun 2022 08:55:19 -0400 (EDT)
X-Delivered-To: linux-mm@kvack.org
Received: from relay.hostedemail.com (smtprelay0017.hostedemail.com
 [216.40.44.17])
	by kanga.kvack.org (Postfix) with ESMTP id 4C1538E020E
	for <linux-mm@kvack.org>; Fri, 24 Jun 2022 08:55:19 -0400 (EDT)
Received: from smtpin15.hostedemail.com (a10.router.float.18 [10.200.18.1])
	by unirelay13.hostedemail.com (Postfix) with ESMTP id 281B960D16
	for <linux-mm@kvack.org>; Fri, 24 Jun 2022 12:55:19 +0000 (UTC)
X-FDA: 79613125158.15.87FECE9
Received: from outbound-smtp17.blacknight.com (outbound-smtp17.blacknight.com
 [46.22.139.234])
	by imf04.hostedemail.com (Postfix) with ESMTP id A59A440004
	for <linux-mm@kvack.org>; Fri, 24 Jun 2022 12:55:18 +0000 (UTC)
Received: from mail.blacknight.com (pemlinmail06.blacknight.ie
 [81.17.255.152])
	by outbound-smtp17.blacknight.com (Postfix) with ESMTPS id 6A8121C3DB8
	for <linux-mm@kvack.org>; Fri, 24 Jun 2022 13:55:17 +0100 (IST)
Received: (qmail 8258 invoked from network); 24 Jun 2022 12:55:17 -0000
Received: from unknown (HELO morpheus.112glenside.lan)
 (mgorman@techsingularity.net@[84.203.198.246])
  by 81.17.254.9 with ESMTPA; 24 Jun 2022 12:55:16 -0000
From: Mel Gorman <mgorman@techsingularity.net>
To: Andrew Morton <akpm@linux-foundation.org>
Cc: Nicolas Saenz Julienne <nsaenzju@redhat.com>,
	Marcelo Tosatti <mtosatti@redhat.com>,
	Vlastimil Babka <vbabka@suse.cz>,
	Michal Hocko <mhocko@kernel.org>,
	Hugh Dickins <hughd@google.com>,
	Yu Zhao <yuzhao@google.com>,
	Marek Szyprowski <m.szyprowski@samsung.com>,
	LKML <linux-kernel@vger.kernel.org>,
	Linux-MM <linux-mm@kvack.org>,
	Mel Gorman <mgorman@techsingularity.net>
Subject: [PATCH 4/7] mm/page_alloc: Remove mistaken page == NULL check in
 rmqueue
Date: Fri, 24 Jun 2022 13:54:20 +0100
Message-Id: <20220624125423.6126-5-mgorman@techsingularity.net>
X-Mailer: git-send-email 2.35.3
In-Reply-To: <20220624125423.6126-1-mgorman@techsingularity.net>
References: <20220624125423.6126-1-mgorman@techsingularity.net>
MIME-Version: 1.0
ARC-Seal: i=1; s=arc-20220608; d=hostedemail.com; t=1656075318; a=rsa-sha256;
	cv=none;
	b=z5hiIqa/3GK2YBLoXUEFkiBm0kXuLI+t1nhYiHDewkLYsyx350OyMIppHlFHclzHf/oUhJ
	MJcYCZkWS0SJDWf1UVbLNlSGFYssNB9dvLVNqCKgGmxMkNt8ZgJkq3i/1NdIaH8FqjKPep
	mIifZdeQg471kzYyAJ4dlUwpOi+mvp8=
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed;
 d=hostedemail.com;
	s=arc-20220608; t=1656075318;
	h=from:from:sender:reply-to:subject:subject:date:date:
	 message-id:message-id:to:to:cc:cc:mime-version:mime-version:
	 content-type:content-transfer-encoding:content-transfer-encoding:
	 in-reply-to:in-reply-to:references:references;
	bh=CpKbpBkRPx5i5h1XOAeKyIGn1KxXP1MbOZRyma+4oAc=;
	b=fPKNWTyPUgDLH4SkS+V3IOpU3RqZLDLxZKwPRid+v8utiljcOmYIzs+qYJoGZNmCqD9pQn
	x1Hc6l5IR0bd8fN4yQMaBTOCe2UnBTI1e53yM9hkFCpo+o7091Ukgv3YAhUu+YfSg2HRmC
	R6yZ4XegcHSxER9SU1XvAvVuaPN2MoI=
ARC-Authentication-Results: i=1;
	imf04.hostedemail.com;
	dkim=none;
	dmarc=none;
	spf=pass (imf04.hostedemail.com: domain of mgorman@techsingularity.net
 designates 46.22.139.234 as permitted sender)
 smtp.mailfrom=mgorman@techsingularity.net
X-Stat-Signature: jj39hzwtwpfpdfmjxt7i5jnrhun1xoes
X-Rspamd-Queue-Id: A59A440004
Authentication-Results: imf04.hostedemail.com;
	dkim=none;
	dmarc=none;
	spf=pass (imf04.hostedemail.com: domain of mgorman@techsingularity.net
 designates 46.22.139.234 as permitted sender)
 smtp.mailfrom=mgorman@techsingularity.net
X-Rspam-User: 
X-Rspamd-Server: rspam04
X-HE-Tag: 1656075318-235061
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

If a page allocation fails, the ZONE_BOOSTER_WATERMARK should be tested,
cleared and kswapd woken whether the allocation attempt was via the PCP
or directly via the buddy list.

Remove the page == NULL so the ZONE_BOOSTED_WATERMARK bit is checked
unconditionally. As it is unlikely that ZONE_BOOSTED_WATERMARK is set,
mark the branch accordingly.

Signed-off-by: Mel Gorman <mgorman@techsingularity.net>
Acked-by: Vlastimil Babka <vbabka@suse.cz>
---
 mm/page_alloc.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/mm/page_alloc.c b/mm/page_alloc.c
index 111111111..111111111 100644
--- a/mm/page_alloc.c
+++ b/mm/page_alloc.c
@@ -3782,12 +3782,10 @@ struct page *rmqueue(struct zone *preferred_zone,
 
 	page = rmqueue_buddy(preferred_zone, zone, order, alloc_flags,
 							migratetype);
-	if (unlikely(!page))
-		return NULL;
 
 out:
 	/* Separate test+clear to avoid unnecessary atomics */
-	if (test_bit(ZONE_BOOSTED_WATERMARK, &zone->flags)) {
+	if (unlikely(test_bit(ZONE_BOOSTED_WATERMARK, &zone->flags))) {
 		clear_bit(ZONE_BOOSTED_WATERMARK, &zone->flags);
 		wakeup_kswapd(zone, 0, 0, zone_idx(zone));
 	}

